astromon/cross_match.py

```diff
@@ -468,26 +699,87 @@ def _simple_cross_match(
         keys=['obsid', 'x_id'],
         table_names=['xray', 'cat']
     )
-    assert np.all(matches['id_xray'] == matches['x_id'])
-    matches.rename_column('id_cat', 'c_id')
 
-    matches['dz'] = matches['z_angle_xray'] - matches['z_angle_cat']
-    matches['dy'] = matches['y_angle_xray'] - matches['y_angle_cat']
+    matches['dz'] = matches['x_z_angle'] - matches['c_z_angle']
+    matches['dy'] = matches['x_y_angle'] - matches['c_y_angle']
     matches['dr'] = np.sqrt(matches['dy']**2 + matches['dz']**2)
     matches['cat_order'] = 200
     for i, k in enumerate(catalogs):
-        matches[matches['catalog'] == k]['cat_order'] = i
+        matches['cat_order'][matches['catalog'] == k] = i
 
-    matches = matches[matches['r_angle'] < r_angle]
+    matches = matches[
+        ((matches['grating'] == 'NONE') & (matches['r_angle'] < r_angle))
+        | ((matches['grating'] != 'NONE') & (matches['r_angle'] < r_angle_grating))
+    ]
+    matches = matches[matches['dr'] < dr]
 
     if len(matches) == 0:
         return []
```

Comment:
Why are you turning the 0-length table that has all the right columns into a list? Better to have a consistent return type.

Reply:
This was changed in a recent [commit](https://github.com/sot/astromon/pull/17/commits/c5fc7b9e0e30cecf1165e78283eb6e9e7113fd61). In order to be fully consistent, I had to make a few other changes.